pkg/discovery/monitoring/kubelet/kubelet_monitor.go

```diff
@@ -126,20 +135,42 @@ func (m *KubeletMonitor) scrapeKubelet(node *api.Node) {
 		m.metricSink.AddNewMetricEntries(cacheUsedMetric)
 	}
 
-	m.parseNodeStats(summary.Node)
-	m.parsePodStats(summary.Pods)
+	currentMilliSec := time.Now().UnixNano() / int64(time.Millisecond)
+	m.parseNodeStats(summary.Node, currentMilliSec)
+	m.parsePodStats(summary.Pods, currentMilliSec)
 
 	glog.V(4).Infof("Finished scrape node %s.", node.Name)
 }
 
+// scrapeKubeletForSampling collects resource usage data samples for node, and containers and pods in the node.
+func (m *KubeletMonitor) scrapeKubeletForSampling(node *api.Node) {
+	kc := m.kubeletClient
+	ip, err := util.GetNodeIPForMonitor(node, types.KubeletSource)
+	if err != nil {
+		glog.Errorf("Failed to get resource metrics summary from %s: %s", node.Name, err)
+		return
+	}
+	// get summary information about the given node and the pods running on it.
+	summary, err := kc.GetSummary(ip)
+	if err != nil {
+		glog.Errorf("Failed to get resource metrics summary from %s: %s", node.Name, err)
+		return
+	}
+
+	currentMilliSec := time.Now().UnixNano() / int64(time.Millisecond)
+	m.parseNodeStats(summary.Node, currentMilliSec)
```

Comment:
I thought the only data we did want to discover in this time based sampling is the container utilisation data useful for driving actions via constainerSpecs. Am I missing something?

Reply:
Current resource peak and used values are always the same. We also want to discover more usage data samples for pod and node, and then use max value as "peak" and average as "used" so as to better extract and show the peak number.